### ISSUE TITLE
Response transformer plugin causes all requests to timeout

### ISSUE THREAD

**Issue by ThorArakis** [Mashape/kong#1283]
When I have a response transformer plugin active on an endpoint it seems to cause the endpoint to stop responding, and all requests through that endpoint time out. When I disable the plugin the request start working again.

This appears to happen with all of the different configurations that I have tried on the response transformer plugin, and also happens when there is a single plugin or multiple plugins enabled on the endpoint.

plugin configuration:
```
{
  "data": [
    {
      "api_id": "<id_here>",
      "id": "<id_here>",
      "created_at": 1464822763000,
      "enabled": true,
      "name": "response-transformer",
      "config": {
        "remove": {
          "json": {},
          "headers": {}
        },
        "add": {
          "json": {},
          "headers": {}
        },
        "replace": {
          "json": {},
          "headers": {}
        },
        "append": {
          "json": {},
          "headers": {}
        }
      }
    }
  ],
  "total": 1
}
```

**Comment by thefosk** [comment]
Is the response a streaming response?

**Comment by ThorArakis** [comment]
Nope, just a basic call through to httpbin

```
{
  "upstream_url": "http://httpbin.org/get",
  "strip_request_path": true,
  "request_path": "/test",
  "id": "2f1c6f37-2a28-48ad-bb18-3187cc8bf4d2",
  "created_at": 1464822704000,
  "preserve_host": false,
  "name": "test"
}
```

**Comment by thefosk** [comment]
Can you try to replicate this with the new 0.8.3 release?

**Comment by Tieske** [comment]
Has been fixed by #1263 @ThorArakis please confirm.

As there is no body transform, the content-length header would be left unchanged by the plugin. Yet, the body would be parsed, and then re-jsonified. Causing a different length because of whitespace differences between the original and the re-jsonified body.

**Comment by ThorArakis** [comment]
I just pulled the latest from master and can confirm that the response transformer no longer kills the request! Thanks!
